src/main/java/seedu/module/model/ModuleManager.java

```diff
@@ -53,22 +63,68 @@ public static boolean moduleIsValid(String module) {
      */
     public static void insertTaskToMapping(Module module, Task task) {
         if (mappingOfModulesToTasks == null) {
-            clearMapping();
+            rebuildMapping();
         }
-        if (mappingOfModulesToTasks.containsKey(module.toString())) {
-            List<Task> newList = mappingOfModulesToTasks.get(module.toString());
+        if (mappingOfModulesToTasks.containsKey(module)) {
+            List<Task> newList = mappingOfModulesToTasks.get(module);
             //must ensure Module exists in the listOfValidModules
             newList.add(task);
-            module.incrementWorkload(task.getWorkload());
-            mappingOfModulesToTasks.put(module.toString(), newList);
+            mappingOfModulesToTasks.put(module, newList);
         } else {
             List<Task> newList = new ArrayList<>();
             newList.add(task);
-            module.incrementWorkload(task.getWorkload());
-            mappingOfModulesToTasks.put(module.toString(), newList);
+            mappingOfModulesToTasks.put(module, newList);
+            moduleWorkLoadDistribution.put(module, task.getWorkload().getWorkloadLevel());
+        }
+        putIntoCorrectWorkloadDistribution(module, task);
+        setExistingModuleList();
+        setModulePieChartData();
+    }
+
+    /**
+     * Modifies the workload distribution after adding the task.
+     *
+     * @param module the corresponding module of the task
+     * @param task the task that need to be inserted into mapping book.
+     */
+    public static void putIntoCorrectWorkloadDistribution(Module module, Task task) {
+        int workloadLevel = task.getWorkload().getWorkloadLevel();
+        moduleWorkLoadDistribution.put(module,
+            moduleWorkLoadDistribution.get(module) + workloadLevel);
+        switch(workloadLevel) {
+        case 1:
```

Comment:
just a minor suggestion, to change 1,2,3 to low,medium, high for clearer communication through code

Reply:
Resolved.